src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedLogicalPlanPrinterImpl2.java

```diff
@@ -48,11 +71,43 @@ public String getIndentLevelString(final int planNumber, final int planLevel, fi
 	 */
 	public void planWalk( final LogicalPlan plan, final int planNumber, final int planLevel, final int numberOfSiblings, final PrintStream out) {
 		final String indentLevelString = getIndentLevelString(planNumber, planLevel, numberOfSiblings);
-		out.append( indentLevelString + plan.getRootOperator().toString() );
-		out.append( System.lineSeparator() );
+		final LogicalOperator rootOp = plan.getRootOperator();
+		if ( rootOp instanceof LogicalOpMultiwayJoin ) {
+			printOperatorInfoMultiwayJoin( (LogicalOpMultiwayJoin) rootOp, out, indentLevelString );
+		}
+		else if (rootOp instanceof LogicalOpMultiwayUnion) {
+			printOperatorInfoMultiwayUnion( (LogicalOpMultiwayUnion) rootOp, out, indentLevelString );
+		}
+		else if (rootOp instanceof LogicalOpRequest) {
+			String indentLevelStringForOpDetail = getIndentLevelStringForDetail(planNumber, planLevel, numberOfSiblings);
+			printOperatorInfoForRequest( (LogicalOpRequest) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
 		for ( int i = 0; i < plan.numberOfSubPlans(); ++i ) {
 			planWalk( plan.getSubPlan(i), i, planLevel+1, plan.numberOfSubPlans(), out );
 		}
 	}
-
+	
+	protected void printOperatorInfoForRequest( final LogicalOpRequest op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		final FederationMember fm = op.getFederationMember();
+		final DataRetrievalRequest req =op.getRequest();
+		out.append( indentLevelString + "req" + " (" + op.getID() + ")" );
+		out.append( System.lineSeparator() );
+		out.append( indentLevelStringForOpDetail + "  " + "- fm (" + fm.getInterface().toString() + ")" );
```

Comment:
As we have several types of operators that contain a federation member as one of their parameters, it makes sense to print this particular element of detailed information always in the same way. Therefore, please create a separate helper function for putting together this string and, then, call this function here. And later you can also reuse this function for the other types of operators.

Reply:
Just take bgpAdd as an example, 
```
return "> bgpAdd" +
		"[" + codeOfBGP + ", "+ codeOfFm + "]"+
		" ( "
		+ bgp.toString()
		+ ", "
		+ fm.getInterface().toString()
		+ " )";
```
Should we print like the following way?
├bgpAdd
        - codeOfBGP, codeOfFm [-1530180337, -34331270] 
	- fm (SPARQL endpoint at https://query.wikidata.org/sparql )
	- pattern (bgp ?cc  <http://www.w3.org/2000/01/rdf-schema#label>  ?o . )